app/redux/reducers/SetsReducer.js

```diff
@@ -217,6 +229,19 @@ const saveHistorySet = (state, action) => {
     return Object.assign({}, state, stateChanges);
 };
 
+// DELETE_HISTORY_SET
+
+const deleteHistorySet = (state, action) => {
+    return Object.assign({}, state, {
+        historyData: Object.keys(state.historyData).reduce((result, setID) => {
+            if (setID !== action.setID) {
+                result[setID] = state.historyData[setID];
+            }
+            return result;
+        }, {})
```

Comment:
- [ ] Does this entire part work? I've not used reduce and keys in this combination before, it's interesting. But it's still referencing the original historyData[setID] rather than making a copy, isn't that going to cause issues in redux?